node/peers.js

```diff
@@ -136,12 +136,6 @@ TChannelPeers.prototype.delete = function del(hostPort) {
     var self = this;
     var peer = self._map[hostPort];
     delete self._map[hostPort];
-    if (self.subChannels) {
```

Comment:
This does not match the `add()` semantics;

If you `add()` a peer to subchannel A; we add it to top channel.

What it sounds like we want for sub channel is;

 - if you `remove()` a peer from a sub channel AND no other sub channel has that peer; remove it from the top channel.